boot/seal.go

```diff
@@ -188,7 +188,7 @@ func resealKeyToModeenv(rootdir string, model *asserts.Model, modeenv *Modeenv)
 
 	pbc := toPredictableBootChains(append(runModeBootChains, recoveryBootChains...))
 
-	ok, nextCount, err := isResealNeeded(pbc, rootdir)
+	ok, nextCount, err := isResealNeeded(pbc, rootdir, expectReseal)
```

Comment:
I think the var name `ok` here is slightly confusing in terms of Go's typical usage of `ok` to mean that something is "ok", when this function is called "isResealNeeded", would suggest something like this:

```suggestion
	needed, nextCount, err := isResealNeeded(pbc, rootdir, expectReseal)
```

Reply:
Agree with Ian, "needed" could be better here.